scico/solver.py

```diff
@@ -577,3 +577,371 @@ def golden(
+            W: Matrix :math:`W`.
+            cho_factor: Flag indicating whether to use Cholesky
+                (``True``) or LU (``False``) factorization.
+            lower: Flag indicating whether lower (``True``) or upper
+                (``False``) triangular factorization should be computed.
+                Only relevant to Cholesky factorization.
+            check_finite: Flag indicating whether the input array should
+                be checked for ``Inf`` and ``NaN`` values.
+        """
+        if isinstance(A, MatrixOperator):
+            A = A.to_array()
+        if isinstance(D, MatrixOperator):
+            D = D.to_array()
+        elif isinstance(D, Diagonal):
+            D = D.diagonal
+        if W is None:
+            W = snp.ones(A.shape[0], dtype=A.dtype)
+        elif isinstance(W, Diagonal):
+            W = W.diagonal
+        self.A = A
+        self.D = D
+        self.W = W
+        self.cho_factor = cho_factor
+        self.lower = lower
+        self.check_finite = check_finite
+
+        assert isinstance(W, Array)
+        N, M = A.shape
+        if N < M and D.ndim == 1:
+            G = snp.diag(1.0 / W) + A @ (A.T.conj() / D[:, snp.newaxis])
+        else:
+            if D.ndim == 1:
+                G = A.T.conj() @ (W[:, snp.newaxis] * A) + snp.diag(D)
+            else:
+                G = A.T.conj() @ (W[:, snp.newaxis] * A) + D
+
+        if cho_factor:
+            c, lower = jsl.cho_factor(G, lower=lower, check_finite=check_finite)
+            self.factor = (c, lower)
+        else:
+            lu, piv = jsl.lu_factor(G, check_finite=check_finite)
+            self.factor = (lu, piv)
+
+    def solve(self, b: Array, check_finite: Optional[bool] = None) -> Array:
+        r"""Solve the linear system.
+
+        Solve the linear system with right hand side :math:`\mb{b}` (`b`
+        is a vector) or :math:`B` (`b` is a 2d array).
+
+        Args:
+           b: Vector :math:`\mathbf{b}` or matrix :math:`B`.
+           check_finite: Flag indicating whether the input array should
+               be checked for ``Inf`` and ``NaN`` values. If ``None``,
+               use the value selected on initialization.
+
+        Returns:
+          Solution to the linear system.
+        """
+        if check_finite is None:
+            check_finite = self.check_finite
+        if self.cho_factor:
+            fact_solve = lambda x: jsl.cho_solve(self.factor, x, check_finite=check_finite)
+        else:
+            fact_solve = lambda x: jsl.lu_solve(self.factor, x, trans=0, check_finite=check_finite)
+
+        if b.ndim == 1:
+            D = self.D
+        else:
+            D = self.D[:, snp.newaxis]
+        N, M = self.A.shape
+        if N < M and self.D.ndim == 1:
+            w = fact_solve(self.A @ (b / D))
+            x = (b - (self.A.T.conj() @ w)) / D
+        else:
+            x = fact_solve(b)
+
+        return x
+
+    def accuracy(self, x: Array, b: Array) -> float:
```

Comment:
Perhaps `check_accuracy` or `compute_accuracy`?

Reply:
I'm inclined to leave this as is since the current name seems consistent with naming elsewhere, e.g. some linops have a `norm` method (not `compute_norm`).